Description: Fix msfvenom
 "msfvenom -l all" and "msfvenom -l encrypt" fail with:
 "Traceback (most recent call last):
	3: from /usr/bin/msfvenom:363:in `<main>'
	2: from /usr/bin/msfvenom:363:in `each'
	1: from /usr/bin/msfvenom:385:in `block in <main>'
 /usr/bin/msfvenom:257:in `dump_encrypt': undefined method
 `encryption_formats' for Msf::Simple::Buffer:Module (NoMethodError)"
 I picked up the encryption_formats definition from master branch and
  add it in file lib/msf/base/simple/buffer.rb (currently releases are
  base on branches 4.x)
Origin: upstream, https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/base/simple/buffer.rb
Bug: https://github.com/rapid7/metasploit-framework/issues/10152
Last-Update: 2018-06-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/lib/msf/base/simple/buffer.rb
+++ b/lib/msf/base/simple/buffer.rb
@@ -120,6 +120,15 @@ module Buffer
     ]
   end
 
+    def self.encryption_formats
+    [
+      'xor',
+      'base64',
+      'aes256',
+      'rc4'
+    ]
+  end
+
 end
 
 end
